src/proposer/proposer.cpp

```diff
@@ -29,15 +30,83 @@
 
 namespace proposer {
 
-class ProposerStub : public Proposer {
+bool GenerateBlock(staking::ActiveChain &active_chain,
+                   staking::TransactionPicker &transaction_picker,
+                   proposer::BlockBuilder &block_builder,
+                   proposer::Logic &logic,
+                   CWallet *wallet,
+                   const CBlockIndex &tip,
+                   const staking::CoinSet &coins,
+                   const boost::optional<CScript> &coinbase_script,
+                   std::shared_ptr<const CBlock> &block_out) {
+
+  auto &wallet_ext = wallet->GetWalletExtension();
+  const auto wallet_name = wallet->GetName();
+
+  const boost::optional<EligibleCoin> &winning_ticket = logic.TryPropose(coins);
```

Comment:
This should probably not be a reference.

Reply:
It was a reference before already, why not?

Reply:
I also think it should not be a reference. I believe this is a mistake introduced by me in an earlier pull request. The declaration of `TryPropose` reads:

```C++
virtual boost::optional<proposer::EligibleCoin> TryPropose(const staking::CoinSet &) = 0;
```

i.e. it returns by value, not by reference.

Reply:
I was under the impression that `LogicImpl::TryPropose` returned a temporary variable -- but if it worked before, then I stand corrected.